workspace.go

```diff
@@ -203,21 +203,51 @@ func (w *Workspace) OpenFile(absFilepath, text string) error {
 	w.Loader.done = false
 	w.Loader.stateChange <- absPath
 
+	asc := flattenAscendants(n)
+
+	for _, p1 := range asc {
+		p1.loadState = unloaded
+		p1.ResetChecker()
+		w.Loader.stateChange <- p1.absPath
+	}
+
 	w.log.Debugf("Shadowed file '%s'\n", absFilepath)
 
 	return nil
 }
 
 // ReplaceFile replaces the entire contents of an opened file
-func (w *Workspace) ReplaceFile(absPath, text string) error {
-	_, ok := w.Loader.openedFiles[absPath]
+func (w *Workspace) ReplaceFile(absFilepath, text string) error {
+	_, ok := w.Loader.openedFiles[absFilepath]
 	if !ok {
-		return fmt.Errorf("File %s is not opened", absPath)
+		return fmt.Errorf("File %s is not opened", absFilepath)
 	}
 
 	// Replace the entire document
 	buf := rope.CreateRope(text)
-	w.Loader.openedFiles[absPath] = buf
+	w.Loader.openedFiles[absFilepath] = buf
+
```

Comment:
Resetting the checker for current and ascendant packages can be refactored into a helper method.